### PR TITLE
SDK-2028: Add support for providing a flag to allow non latin documen…

### PR DESCRIPTION
**Pull request by artem-yoti**
…ts at session creation

### CODE REVIEW

File: yoti-sdk-api/src/main/java/com/yoti/api/client/docs/session/create/filters/OrthogonalRestrictionsFilter.java [hunk 1]
```diff
@@ -36,31 +36,67 @@ public static class Builder {
...
+         * @param countryCodes the country codes
+         * @return the builder
+         */
         public Builder withWhitelistedCountries(List<String> countryCodes) {
             countryRestriction = new CountryRestriction(DocScanConstants.INCLUSION_WHITELIST, countryCodes);
             return this;
         }
 
+        /**
+         * Sets the not allowed countries by country codes
```
Comment (MrBurtyyy) [inline yoti-sdk-api/src/main/java/com/yoti/api/client/docs/session/create/filters/OrthogonalRestrictionsFilter.java:55]:
```suggestion
         * Sets the disallowed documents by country code
```

File: yoti-sdk-api/src/main/java/com/yoti/api/client/docs/session/create/filters/OrthogonalRestrictionsFilter.java [hunk 2]
```diff
@@ -36,31 +36,67 @@ public static class Builder {
...
+         * @param documentTypes the document types
+         * @return the builder
+         */
         public Builder withWhitelistedDocumentTypes(List<String> documentTypes) {
             typeRestriction = new TypeRestriction(DocScanConstants.INCLUSION_WHITELIST, documentTypes);
             return this;
         }
 
+        /**
+         * Sets the not allowed document types
```
Comment (MrBurtyyy) [inline yoti-sdk-api/src/main/java/com/yoti/api/client/docs/session/create/filters/OrthogonalRestrictionsFilter.java:77]:
```suggestion
         * Sets the disallowed document types
```

File: yoti-sdk-api/src/main/java/com/yoti/api/client/docs/session/create/filters/OrthogonalRestrictionsFilter.java [hunk 3]
```diff
@@ -36,31 +36,67 @@ public static class Builder {
...
+         * @param documentTypes the document types
+         * @return the builder
+         */
         public Builder withBlacklistedDocumentTypes(List<String> documentTypes) {
             typeRestriction = new TypeRestriction(DocScanConstants.INCLUSION_BLACKLIST, documentTypes);
             return this;
         }
 
+        /**
+         * Sets the flag whether allow non latin documents or not
```
Comment (MrBurtyyy) [inline yoti-sdk-api/src/main/java/com/yoti/api/client/docs/session/create/filters/OrthogonalRestrictionsFilter.java:88]:
```suggestion
         * Sets the flag whether to allow non latin documents or not
```

File: yoti-sdk-api/src/test/java/com/yoti/api/client/docs/session/create/filters/DocumentRestrictionsFilterTest.java [hunk 4]
```diff
@@ -92,4 +92,14 @@ public void shouldAcceptPreBuiltDocumentRestriction() {
         assertThat(result.getDocuments(), containsInAnyOrder(documentRestrictionMock));
     }
 
+    @Test
```
Comment (MrBurtyyy) [inline yoti-sdk-api/src/test/java/com/yoti/api/client/docs/session/create/filters/DocumentRestrictionsFilterTest.java:95]: Can we add a test to ensure, that if no value is provided (i.e. the `with` method is not called), that the value passed from the builder to the domain model object is `null`?

File: yoti-sdk-api/src/test/java/com/yoti/api/client/docs/session/create/filters/OrthogonalRestrictionsFilterTest.java [hunk 5]
```diff
@@ -79,4 +79,13 @@ public void shouldBuildBlacklistedDocumentTypeRestriction() {
         assertThat(result.getTypeRestriction().getDocumentTypes(), containsInAnyOrder("PASSPORT", "DRIVING_LICENCE"));
     }
 
+    @Test
```
Comment (MrBurtyyy) [inline yoti-sdk-api/src/test/java/com/yoti/api/client/docs/session/create/filters/OrthogonalRestrictionsFilterTest.java:82]: Same here please (we want to ensure that the SDK is not providing default values)

**Review by MrBurtyyy — CHANGES_REQUESTED** [review]

File: yoti-sdk-api/src/main/java/com/yoti/api/client/docs/session/create/filters/OrthogonalRestrictionsFilter.java (hunk 1 above)
Comment (artem-yoti) [reply yoti-sdk-api/src/main/java/com/yoti/api/client/docs/session/create/filters/OrthogonalRestrictionsFilter.java]: Amended

File: yoti-sdk-api/src/main/java/com/yoti/api/client/docs/session/create/filters/OrthogonalRestrictionsFilter.java (hunk 2 above)
Comment (artem-yoti) [reply yoti-sdk-api/src/main/java/com/yoti/api/client/docs/session/create/filters/OrthogonalRestrictionsFilter.java]: Amended

File: yoti-sdk-api/src/test/java/com/yoti/api/client/docs/session/create/filters/DocumentRestrictionsFilterTest.java (hunk 4 above)
Comment (artem-yoti) [reply yoti-sdk-api/src/test/java/com/yoti/api/client/docs/session/create/filters/DocumentRestrictionsFilterTest.java:96]: Added a separate test to prove this

**Review by MrBurtyyy — APPROVED** [review]

**Pull request merged**